File: src/server/chat/src/main/java/smilegate/plop/chat/domain/room/Member.java
package smilegate.plop.chat.domain.room;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@NoArgsConstructor
@Setter
@Getter
public class Member {

    private String userId;
    private String lastReadMsgId;
    private LocalDateTime enteredAt;

    @Builder
    public Member(String userId, LocalDateTime enteredAt) {
        this.userId = userId;
        this.enteredAt = enteredAt;
    }
}

File: src/mobile/android/app/src/main/java/com/plop/plopmessenger/data/dto/response/Member.kt
package com.plop.plopmessenger.data.dto.response


import com.plop.plopmessenger.data.local.entity.Member as MemberEntity
import java.time.LocalDateTime

data class Member(
    val enteredAt: String,
    val lastReadMsgId: String?,
    val userId: String
)

fun Member.toMember(roomId: String) = MemberEntity(
    chatroomId = roomId,
    memberId = this.userId,
    nickname = "",
    profileImg = "",
    readMessage = this.lastReadMsgId
)